src/API.jl

```diff
@@ -939,29 +939,34 @@ function precompile(ctx::Context; internal_call::Bool=false)
         ]
     end    
     
-    was_processed = Dict{Base.PkgId,Base.Event}()
+    was_processed = Dict{Base.PkgId,Bool}()
```

Comment:
With this PR, `was_processed` Events are just being used for their underlying Bool, so I just switched it to a Bool